projects/packages/search/src/dashboard/components/pages/sections/plan-usage-section.jsx

```diff
@@ -24,6 +24,207 @@ const usageInfoFromAPIData = apiData => {
+		'jetpack-search-pkg'
+	);
+	const messageOverLimitThree = __(
+		'You’ve exceeded the number of search requests available on the free plan for three consecutive months.',
+		'jetpack-search-pkg'
+	);
+	// Note: The API doesn't give us enough data to determine the near cases.
+	// Start by containing the index.
+	let index = apiData.currentUsage.months_over_plan_requests_limit;
+	if ( index < 0 ) {
+		index = 0;
+	} else if ( index > 3 ) {
+		index = 3;
+	}
+	// Pick the appropriate message.
+	const message = [
+		messageNearLimitOne,
+		messageOverLimitOne,
+		messageOverLimitTwo,
+		messageOverLimitThree,
+		messageNearLimitTwo, // not used
+		messageNearLimitThree, // not used
+	][ index ];
+	// Possible CTA messages here.
+	const ctaUpgradeToAvoid = __(
+		'Upgrade now to increase your monthly request limit and avoid interruption.',
+		'jetpack-search-pkg'
+	);
+	const ctaUpgradeToContinue = __(
+		'Upgrade now to contineu using Jetpack Search.',
+		'jetpack-search-pkg'
+	);
+	// Pick the appropriate CTA.
+	let cta = ctaUpgradeToAvoid;
+	if ( apiData.currentUsage.months_over_plan_requests_limit >= 3 ) {
+		cta = ctaUpgradeToContinue;
+	}
+
+	return { description: message, cta: cta };
+};
+
+const upgradeMessageNoOverage = apiData => {
+	// Always returns a valid message.
+	const message = __(
+		'Do you want to increase your site records and search requests?.',
+		'jetpack-search-pkg'
+	);
+	const cta = __( 'Upgrade now and avoid any future interruption!', 'jetpack-search-pkg' );
+	return { description: message, cta: cta };
+};
+
+const upgradeMessageBoth = apiData => {
+	// Return a valid message or null.
+	const isBothScenario =
+		apiData.currentUsage.upgrade_reason.records && apiData.currentUsage.upgrade_reason.requests;
+	if ( ! isBothScenario ) {
+		return null;
+	}
+	// Determine upgrade message.
+	const messageNearLimit = __(
+		'You’re close to exceeding the number of site records and search requests available for the free plan.',
+		'jetpack-search-pkg'
+	);
+	const messageOverLimit = __(
+		'You’ve exceeded the number of site records and search requests available for the free plan.',
+		'jetpack-search-pkg'
+	);
+	// If the "months over" is zero, use the near messaging.
+	let message = messageOverLimit;
+	if ( apiData.currentUsage.months_over_plan_records_limit === 0 ) {
+		message = messageNearLimit;
+	}
+	// Add the shared CTA.
+	const sharedCTA = __(
+		'Upgrade now to increase your limits and avoid interruption!',
+		'jetpack-search-pkg'
+	);
+	return { description: message, cta: sharedCTA };
+};
+
+const upgradeMessageFromAPIData = apiData => {
+	// What's the data we're working with?
+	// apiData.currentUsage.must_upgrade
+	// apiData.currentUsage.upgrade_reason.records
+	// apiData.currentUsage.upgrade_reason.requests
+	// apiData.currentUsage.months_over_plan_records_limit
+	// apiData.currentUsage.months_over_plan_requests_limit
```

Comment:
In the future, let's use doc comments for this function to define the `apiData` data structure.